src/Searcher.tsx

```diff
@@ -6,18 +6,11 @@ import { LibraryContainer } from "./components/LibraryContainer";
 import * as LibraryUtilities from "./LibraryUtilities";
 import { SearchBar } from "./components/SearchBar";
 
-type displayMode = "structure" | "list";
-
-interface SearchModeChangedFunc {
-    (inSearchMode: boolean): void;
-}
-
 interface ClearSearchFunc {
     (searchText: string): void;
 }
```

Comment:
Not related to your change but the fact that `ClearSearchFunc` takes a `string` seems a little weird to me.